src/category_theory/opposites.lean

```diff
@@ -6,6 +6,20 @@ Authors: Stephen Morgan, Scott Morrison
 import category_theory.types
 import category_theory.equivalence
 
+/-!
+# Opposite categories
+
+We provide a category instance on `Cᵒᵖ`.
+Here `Cᵒᵖ` is an irreducible typeclass synonym for `C`
+(it is the same one used in the algebra library).
+
+We also provide various mechanisms for constructing opposite morphisms, functors,
+and natural transformations.
+
+Unfortunately, because we do not have a definitional equality `op (op X) = X`,
+there are quite a few variations that are needed in practice.
```

Comment:
Here, do you mean that we must swap `Cᵒᵖ` and `C` in all statements, or is there something subtler going on?

Reply:
No. I am foreshadowing the variations like `functor.left_op` etc. I'm not interested in writing more about this in this PR. Anyone who would like to may contribute commits, of course.